pkg/daemon/daemon.go

```diff
@@ -581,38 +475,45 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 			}
 		}
 	}
-	if dn.platformHelpers.IsOpenshiftCluster() && !dn.platformHelpers.IsHypershift() {
-		if err = dn.getNodeMachinePool(); err != nil {
-			return err
-		}
-	}
-	if reqDrain {
-		if !dn.isNodeDraining() {
+
+	if reqDrain ||
```

Comment:
I had to refactor this to a different function to understand this logic:

```go
if reqDrain {
		returnFromSyncHandler, err := dn.handleDrain(reqReboot)
		if returnFromSyncHandler {
			return err
		}
	}
...

func (dn *Daemon) handleDrain(reqReboot bool) (bool, error) {
	if !utils.ObjectHasAnnotationKey(dn.desiredNodeState, consts.NodeStateDrainAnnotationCurrent) {
		return false, nil
	}

	if utils.ObjectHasAnnotation(dn.desiredNodeState, consts.NodeStateDrainAnnotationCurrent, consts.DrainIdle) {
		return false, nil
	}

	if utils.ObjectHasAnnotation(dn.desiredNodeState, consts.NodeStateDrainAnnotationCurrent, consts.DrainComplete) {
		log.Log.Info("nodeStateSyncHandler(): the node complete the draining")
		return false, nil
	}

	if dn.isNodeDraining() {
		log.Log.Info("nodeStateSyncHandler(): the node is still draining waiting")
		return true, nil
	}

	if dn.disableDrain {
		return false, nil
	}

	if reqReboot {
		log.Log.Info("nodeStateSyncHandler(): apply 'Reboot_Required' label for node")
		if err := dn.applyRequirement(consts.RebootRequired); err != nil {
			return true, err
		}

		log.Log.Info("nodeStateSyncHandler(): apply 'Reboot_Required' label for nodeState")
		if err := utils.AnnotateObject(dn.desiredNodeState,
			consts.NodeStateDrainAnnotation,
			consts.RebootRequired, dn.client); err != nil {
			return true, err
		}

		return true, nil
	}

	log.Log.Info("nodeStateSyncHandler(): apply 'Drain_Required' label for node")
	if err := dn.applyRequirement(consts.DrainRequired); err != nil {
		return true, err
	}

	log.Log.Info("nodeStateSyncHandler(): apply 'Drain_Required' label for nodeState")
	if err := utils.AnnotateObject(dn.desiredNodeState,
		consts.NodeStateDrainAnnotation,
		consts.DrainRequired, dn.client); err != nil {
		return true, err
	}

	return true, nil
}
```

It should be equivalent, unless I introduced coding errors. If it's correct, when the nodeState is DrainComplete, it will continue logging `"nodeStateSyncHandler(): the node complete the draining"`, I'm not getting who moves the nodestate from tha value.

Also, conder to add this refactor or a similar one to improve readability

Reply:
I like the idea of moving this to a different function.
there is one problem that we had before your change
```golang
reqDrain || <code..>
```
after your change we will not access that function on the OR

I will rework this one